components/VM_Arm/src/main.c

```diff
@@ -927,15 +938,38 @@ static int load_linux(vm_t *vm, const char *kernel_name, const char *dtb_name, c
     return 0;
 }
 
-void parse_camkes_linux_attributes(void)
+unsigned int parse_camkes_vm_entry_addr(void)
+{
+    unsigned long addr = strtoul(vm_address_config.kernel_entry_addr, NULL, 0);
+    unsigned long base = strtoul(vm_address_config.ram_base, NULL, 0);
+
+    /* Calculate an entry address if not explicitly set */
+    if (ENTRY_ADDR_NOT_SET == addr) {
+        if (config_set(CONFIG_PLAT_TX1) || config_set(CONFIG_PLAT_TX2) ||
+            config_set(CONFIG_PLAT_QEMU_ARM_VIRT) || config_set(CONFIG_PLAT_ODROIDC2) ||
+            config_set(CONFIG_PLAT_ZYNQMP)) {
+            /* This is likely an aarch64/aarch32 linux difference */
+            addr = base + 0x80000;
+        } else {
+            addr = base + 0x8000;
+        }
+        ZF_LOGW("Warning - %s.vm_address_config.kernel_entry_addr is not set. The entry address \
+                has been set to 0x%lx", get_instance_name(), addr);
+    }
+
+    return addr;
+}
+
+void parse_camkes_vm_attributes(void)
```

Comment:
I thing this is also just an internal helper function for this file, so let's not expose it.
```suggestion
static void parse_camkes_vm_attributes(void)
```

Reply:
Good point. Will make it static.